src/main/java/seedu/address/logic/parser/TypeParsingUtil.java

```diff
@@ -270,7 +270,7 @@ public static Subject parseSubject(String flagName, String input) throws ParseEx
      */
     public static Subject parseSubject(String flagName, String input, boolean isOptional) throws ParseException {
         if (isOptional && parseStrs(flagName, input, true) == null) {
-            return null;
```

Comment:
This will break the current add/edit class, as they are updated via "setIfNotNull" method.

Reply:
Upon reflection, I think it is my fault not articulating the dependence of parser classes on model items. So I want to create an abstract class/interface and make all of the parsable subjects in model extend/implement it. Are you good with that?